packages/runtime/datastore/src/remoteChannelContext.ts

```diff
@@ -227,8 +235,14 @@ export class RemoteChannelContext implements IChannelContext {
      * This returns the GC data retrieved from the base snapshot of this context.
      */
     private async getInitialGCData(): Promise<IGCData | undefined> {
-        // If there is no initial GC data, then the context's GC data will be generated by calling getGCDataInternal
-        // method above.
-        return undefined;
+        if (this.initialGCDetails === undefined) {
+            if (await this.services.objectStorage.contains(gcBlobKey)) {
```

Comment:
Similar feedback here = there are likely a possibility for 20 calls to come into here, there likely needs protection to only read data once (code should not assume that driver does caching if it's important for perf of this layer).